pages/desktop/details.py

```diff
@@ -565,3 +599,105 @@ def ratings_card_summary(self):
+
+        @property
+        def delete_rating_link(self):
+            # waits for the ratings stars to be fully loaded and editable
+            self.wait.until(
+                expected.invisibility_of_element_located(
+                    self._loaded_rating_stars_locator
+                )
+            )
+            return self.find_element(*self._delete_rating_link_locator)
+
+        def delete_confirm_button(self):
+            self.find_element(*self._delete_confirm_button_locator).click()
+            self.wait.until(
+                expected.invisibility_of_element_located(self._review_text_locator)
+            )
+
+        @property
+        def write_a_review(self):
+            return self.find_element(*self._write_review_button_locator)
+
+        def wait_for_rating_form(self):
+            self.wait.until(
+                expected.element_to_be_clickable(self._write_review_button_locator)
+            )
+
+        def review_text_input(self, value):
+            self.find_element(*self._review_textarea_locator).send_keys(value)
+
+        def submit_review(self):
+            self.find_element(*self._submit_review_button_locator).click()
+            self.wait.until(
+                expected.visibility_of_element_located(self._edit_review_link_locator)
+            )
+
+        @property
+        def written_review(self):
+            return self.find_element(*self._review_text_locator)
+
+        @property
+        def cancel_review(self):
+            return self.find_element(*self._cancel_review_write_locator)
+
+        @property
+        def edit_review(self):
+            # waits for the ratings stars to be fully loaded and editable
+            self.wait.until(
+                expected.invisibility_of_element_located(
+                    self._loaded_rating_stars_locator
+                )
+            )
+            return self.find_element(*self._edit_review_link_locator)
+
+        @property
+        def delete_review(self):
+            # waits for the ratings stars to be fully loaded and editable
+            self.wait.until(
+                expected.invisibility_of_element_located(
+                    self._loaded_rating_stars_locator
+                )
+            )
+            return self.find_element(*self._delete_review_link_locator)
+
+        @property
+        def keep_review(self):
+            return self.find_element(*self._keep_review_button_locator)
+
+        @property
+        def review_permalink(self):
+            return self.find_element(*self._review_permalink_locator)
+
+        def all_reviews_link(self):
+            self.find_element(*self._all_reviews_link_locator).click()
+            return Reviews(self.selenium, self.page.base_url).wait_for_page_to_load()
+
+        @property
+        def all_reviews_link_rating_count(self):
+            count = self.find_element(*self._all_reviews_link_locator).text
+            return int(count.split()[2])
+
+        def report_abuse(self):
```

Comment:
This should also have click in the name.
```suggestion
        def click_report_abuse_button(self):
```